src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerChangeService.java

```diff
@@ -0,0 +1,291 @@
+        return changes;
+    }
+
+
+    /**
+     * Identifies and returns changes related to the 'Individual' type managing officers in the
+     * submission.
+     *
+     * <p>This method is responsible for tracking the changes made to the managing officers of
+     * 'Individual' type within the provided submission. Each change is encapsulated within a
+     * {@code ManagingOfficerChange<IndividualManagingOfficer>} object.</p>
+     *
+     * @return A list of changes ({@code ManagingOfficerChange<IndividualManagingOfficer>} objects)
+     * related to the managing officers of 'Individual' type. If no changes are detected, an empty
+     * list is returned.
+     */
+    private List<Change> getIndividualManagingOfficerChange() {
+        var individualManagingOfficers = overseasEntitySubmissionDto.getManagingOfficersIndividual();
+        return individualManagingOfficers
+                .stream()
+                .filter(mo -> mo.getChipsReference() != null)
+                .map(this::covertManagingOfficerIndividualToChange)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    //todo: comment
+    private List<Change> getCorporateManagingOfficerChange() {
+        var corporateManagingOfficers = overseasEntitySubmissionDto.getManagingOfficersCorporate();
+        return corporateManagingOfficers
+                .stream()
+                .filter(mo -> mo.getChipsReference() != null)
+                .map(this::covertManagingOfficerCorporateToChange)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Converts a ManagingOfficerIndividualDto into a ManagingOfficerChange object.
+     *
+     * <p>The method detects changes between the provided managingOfficerIndividualDto and its
+     * corresponding information in the public
+     * and private data.</p>
+     *
+     * @param managingOfficerIndividualDto The ManagingOfficerIndividualDto object to be converted.
+     * @return A ManagingOfficerChange object if changes are detected, otherwise null.
+     */
+    private ManagingOfficerChange<IndividualManagingOfficer> covertManagingOfficerIndividualToChange(
+            ManagingOfficerIndividualDto managingOfficerIndividualDto) {
+        var managingOfficerChange = new IndividualManagingOfficerChange();
+
+        var officer = new IndividualManagingOfficer();
+
+        Pair<CompanyOfficerApi, ManagingOfficerDataApi> publicPrivateMoPair = publicPrivateMo.get(
+                managingOfficerIndividualDto.getChipsReference());
+
+        if (publicPrivateMoPair == null) {
+            ApiLogger.errorContext(SERVICE, NO_PAIR_FOUND, null);
+            return null;
+        }
+        ChangeManager<IndividualManagingOfficer, CompanyOfficerApi, ManagingOfficerDataApi> changeManager = new ChangeManager<>(
+                officer, publicPrivateMoPair);
+
+        if (publicPrivateMoPair.getRight() != null) {
```

Comment:
Please sync to main and see how handling this was changed. We can't process the change without an id. So better to raise an error. Can be a part of 112 - saying no MO can be identified in the DB.